api/indexer/indexer.go

```diff
@@ -8,230 +8,256 @@ import (
 	"encoding/binary"
 	"errors"
 	"fmt"
+	"math"
 	"path/filepath"
-	"sync/atomic"
+	"sync"
 
 	"github.com/ava-labs/avalanchego/database"
 	"github.com/ava-labs/avalanchego/ids"
-	"github.com/ava-labs/avalanchego/utils/wrappers"
 	"github.com/prometheus/client_golang/prometheus"
 
 	"github.com/ava-labs/hypersdk/chain"
 	"github.com/ava-labs/hypersdk/event"
-	"github.com/ava-labs/hypersdk/fees"
-	"github.com/ava-labs/hypersdk/internal/cache"
 	"github.com/ava-labs/hypersdk/internal/pebble"
 )
 
 const maxBlockWindow uint64 = 1_000_000
 
-var errBlockNotFound = errors.New("block not found")
+const (
+	blockEntryByte = iota + 1
+	latestBlockHeightByte
+)
+
+var (
+	blockEntryKeyPrefix  = []byte{blockEntryByte}
+	latestBlockHeightKey = []byte{latestBlockHeightByte}
+)
+
+var (
+	errBlockNotFound          = errors.New("block not found")
+	errTxResultNotFound       = errors.New("transaction result not found")
+	errZeroBlockWindow        = errors.New("indexer configuration of non zero block window is expected")
+	errInvalidBlockWindowSize = errors.New("invalid indexer block window size specified")
+)
 
 var _ event.Subscription[*chain.ExecutedBlock] = (*Indexer)(nil)
 
 type Indexer struct {
 	blockDB            *pebble.Database // height -> block bytes
-	blockIDToHeight    *cache.FIFO[ids.ID, uint64]
-	blockHeightToBlock *cache.FIFO[uint64, *chain.ExecutedBlock]
+	cachedBlocks       []uint64
```

Comment:
This is unnecessary duplicate data since we can always check the length of `blockHeightToBlock` or the other data structures and figure out the latest one.